internal/courses/coursePlans.go

```diff
@@ -0,0 +1,13 @@
+package courses
+
+import "github.com/gzcharleszhang/course-planner/internal/terms"
+
+type TermCoursePlan struct {
+	Term    terms.Term
+	Courses Courses
+}
+
+type CoursePlan struct {
+	CompletedTermPlans []*TermCoursePlan
+	FutureTermPlans    []*TermCoursePlan
```

Comment:
We need a method/func to check pre-reqs between terms. An idea could be to make `TermCoursePlan` an interface. I don't really like how there's a distinction between completed and future term plans, maybe we could abstract that with the interface.

Reply:
there's only 1 slice now, so the distinction between a completed course and a non-completed course is based on if the course is in Student::CourseRecords